theseus/optimizer/nonlinear/gauss_newton.py

```diff
@@ -41,3 +41,29 @@ def __init__(
 
     def compute_delta(self, **kwargs) -> torch.Tensor:
         return self.linear_solver.solve()
+
+    def compute_samples(self, n_samples: int = 10, T: float = 1.0) -> torch.Tensor:
+        delta = self.linear_solver.solve()
+        AtA = self.linear_solver.precision() / T
```

Comment:
Any reason to have a separate `precision` function call since it just return `AtA` which you could directly access here?

Reply:
Moved it to `Linearization` where it is called `hessian_approx()`.